src/main/core/sim_config.rs

```diff
@@ -121,6 +123,14 @@ impl SimConfig {
             }
         }
 
+        // arrange so that hosts with an IP address are at the beginning of the list, to match the
+        // old Shadow behaviour
+        hosts.sort_by(|a, b| match (a.ip_addr.is_some(), b.ip_addr.is_some()) {
```

Comment:
Re this commit overall - is it worth this extra complexity to preserve exactly the previous behavior? It's useful to have done it to verify that this was the only source of deviation, but now that we have I wonder whether it's worth maintaining, or whether we ought to revert this commit? (Maybe something to discuss in the next team meeting; we can always revert this commit later)